### PR TITLE
Email input

### LINKED ISSUE #789: Email input widget
The email input widget will essentially be a base `text-input` with opinionated default properties (such as `type: email`) and a subtly different styling. ie. an `@`/ `mail` icon symbol in the leading section of the input.

The email input should not expose properties such as `min` and `max` and others that do not apply to an email input. Native validation should work for email addresses.

This will expand our widget suite and allow a quick and consistently validated email input for use in dojo applications.

### PR DESCRIPTION
**Pull request by samenz**
**Type:** Feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:** Adds simple email input widget. The email input removes the following text-input properties from it's interface:
- customValidator
- max
- min
- step
- onKeyDown
- onKeyUp
- pattern
- type

**Code Description:**
- Add email input widget
- Add email input widget to examples

Resolves #789

### CODE REVIEW

File: src/email-input/index.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+			onBlur,
+			onFocus,
+			onClick,
+			onOver,
+			onOut,
+			valid,
+			controls
+		} = this.properties;
+
+		return v(
```
Comment (tomdye) [inline src/email-input/index.ts]: Do you need the wrapping `div` ?

File: src/email-input/index.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			'div',
+			{
+				key: 'root',
+				role: 'presentation'
+			},
+			[
+				w(TextInput, {
```
Comment (tomdye) [inline src/email-input/index.ts]: How about doing `<TextInput {...this.properties} type="email" />`

Comment (tomdye) [inline src/email-input/index.ts]: Do you think you could change this to `tsx` ?

File: src/email-input/index.ts (hunk 1 above)
Comment (samenz) [reply src/email-input/index.ts]: Nope, it looked like a pattern that was done in text-input, I can take it out

File: src/email-input/index.ts (hunk 2 above)
Comment (samenz) [reply src/email-input/index.ts]: Yea I can change this to `tsx`, are we trying to write all new widgets in `tsx` cause I'll note that for the next one.

Comment (samenz) [reply src/email-input/index.ts]: Yea I can change this to tsx, are we trying to write all new widgets in tsx cause I'll note that for the next one.

**Pull request merged**